tests/Unit/RequestPaginationTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Tests\Unit;
+
+use GuzzleHttp\Promise\PromiseInterface;
+use Illuminate\Support\LazyCollection;
+use Saloon\Contracts\Response;
+use Saloon\Tests\Fixtures\Connectors\PageRequestPaginatorConnector;
+use Saloon\Tests\Fixtures\Requests\PageGetSuperHeroesRequest;
+
+test('you can yield from a paginator', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $responses = [];
+    $superheroes = [];
+
+    LazyCollection::make($connector->paginate($request))
+        ->each(function (Response $response) use (&$responses, &$superheroes): void {
+            $responses[] = $response;
+            $superheroes = [...$superheroes, ...$response->json('data')];
+        });
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
+});
+
+test('you can continue a paginator from where it left off', function (): void {
+    $connector = new PageRequestPaginatorConnector;
+    $request = new PageGetSuperHeroesRequest;
+
+    $paginator = $connector->paginate($request);
+
+    $responses = [];
+    $superheroes = [];
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+
+        break;
+    }
+
+    expect($responses)->toHaveCount(1)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(5)->each->toBeArray();
+
+    // TODO: See the test skip() message.
+
+    foreach ($paginator as $response) {
+        $responses[] = $response;
+        $superheroes = [...$superheroes, ...$response->json('data')];
+    }
+
+    expect($responses)->toHaveCount(4)->each->toBeInstanceOf(Response::class)
+        ->and($superheroes)->toHaveCount(20)->each->toBeArray();
+})->skip(<<<'REASON'
```

Comment:
Is this test still needed since you removed the serialization now?

Reply:
That test was written for the continuation feature, which is used to break out of loops, and then keep iterating in new ones.
I'm not sure if it'd be used without serialisation, though.
I'll leave that call to you.

Reply:
Oh okay, I'd say keep it then